unified-runtime/source/adapters/level_zero/image.cpp

```diff
@@ -729,15 +729,47 @@ ur_result_t urBindlessImagesMapExternalArrayExp(
 ur_result_t urBindlessImagesMapExternalLinearMemoryExp(
     ur_context_handle_t hContext, ur_device_handle_t hDevice, uint64_t offset,
     uint64_t size, ur_exp_external_mem_handle_t hExternalMem, void **phRetMem) {
-  std::ignore = hContext;
-  std::ignore = hDevice;
-  std::ignore = size;
-  std::ignore = offset;
-  std::ignore = hExternalMem;
-  std::ignore = phRetMem;
-  logger::error("[UR][L0] {} function not implemented!",
-                "{} function not implemented!", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  UR_ASSERT(hContext && hDevice && hExternalMem,
+            UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+  UR_ASSERT(offset && size, UR_RESULT_ERROR_INVALID_BUFFER_SIZE);
+
+  struct ur_ze_external_memory_data *externalMemoryData =
+      reinterpret_cast<ur_ze_external_memory_data *>(hExternalMem);
+  UR_ASSERT(externalMemoryData && externalMemoryData->importExtensionDesc,
+            UR_RESULT_ERROR_INVALID_NULL_POINTER);
+
+  struct ur_ze_external_memory_data *mappedMemory =
+      new struct ur_ze_external_memory_data;
+
+  mappedMemory->importExtensionDesc = externalMemoryData->importExtensionDesc;
+  mappedMemory->type = externalMemoryData->type;
+  mappedMemory->size = size;
+
+  ze_device_mem_alloc_desc_t allocDesc = {};
+  allocDesc.stype = ZE_STRUCTURE_TYPE_DEVICE_MEM_ALLOC_DESC;
+  allocDesc.flags = 0;
```

Comment:
you are not passing the external memory to the pNext so how is this getting imported in the api?